pkg/processor/metricfrequencyprocessor/processor_test.go

```diff
@@ -0,0 +1,211 @@
+	}
+
+	input := createMetrics(resource1metrics, resource2metrics)
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, 1, result.ResourceMetrics().Len())
+	assert.Equal(t, "m2", result.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().At(0).Name())
+}
+
+func TestSelectionWithTwoLibraries(t *testing.T) {
+	ctx := context.Background()
+	sieve := &singleMetricSieve{name: "m1"}
+	processor := &metricsfrequencyprocessor{sieve: sieve}
+
+	resourceMetrics := map[string][]string{
+		"lib-1": {"m1"},
+		"lib-2": {"m1", "m2"},
+	}
+
+	input := createMetrics(resourceMetrics)
+	result, err := processor.ProcessMetrics(ctx, input)
+
+	assert.NoError(t, err)
+	assert.Equal(t, 1, result.ResourceMetrics().Len())
+	assert.Equal(t, 1, result.ResourceMetrics().At(0).InstrumentationLibraryMetrics().Len())
+	assert.Equal(t, "lib-2", result.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).InstrumentationLibrary().Name())
+	assert.Equal(t, 1, result.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().Len())
+	assert.Equal(t, "m2", result.ResourceMetrics().At(0).InstrumentationLibraryMetrics().At(0).Metrics().At(0).Name())
+}
+
+func createGauge() pdata.DoubleGauge {
+	dpSlice := pdata.NewDoubleDataPointSlice()
+	pdata.NewDoubleDataPoint().CopyTo(dpSlice.AppendEmpty())
+
+	gauge := pdata.NewDoubleGauge()
+	dpSlice.CopyTo(gauge.DataPoints())
+
+	return gauge
+}
+
+func createMetric(name string) pdata.Metric {
+	metric := pdata.NewMetric()
+	metric.SetName(name)
+	metric.SetDataType(pdata.MetricDataTypeDoubleGauge)
+	createGauge().CopyTo(metric.DoubleGauge())
+
+	return metric
+}
+
+func createInstrumentedLibrary(name string) pdata.InstrumentationLibrary {
+	library := pdata.NewInstrumentationLibrary()
+	library.SetName(name)
+	library.SetVersion("-")
+
+	return library
+}
+
+func createIlm(name string, metricNames []string) pdata.InstrumentationLibraryMetrics {
+	ilm := pdata.NewInstrumentationLibraryMetrics()
+	createInstrumentedLibrary(name).CopyTo(ilm.InstrumentationLibrary())
+	for _, metricName := range metricNames {
+		createMetric(metricName).CopyTo(ilm.Metrics().AppendEmpty())
+	}
+
+	return ilm
+}
+
+func createRm(metricsPerLibrary map[string][]string) pdata.ResourceMetrics {
+	rm := pdata.NewResourceMetrics()
+	keys := keySliceButDifferent(metricsPerLibrary)
+	sort.Strings(keys)
+	pdata.NewResource().CopyTo(rm.Resource())
+	for _, key := range keys {
+		library := key
+		metrics := metricsPerLibrary[library]
+		createIlm(library, metrics).CopyTo(rm.InstrumentationLibraryMetrics().AppendEmpty())
+	}
+
+	return rm
+}
+
+func createMetrics(metricsPerLibraryArgs ...map[string][]string) pdata.Metrics {
+	metrics := pdata.NewMetrics()
+	for _, metricsPerLibrary := range metricsPerLibraryArgs {
+		createRm(metricsPerLibrary).CopyTo(metrics.ResourceMetrics().AppendEmpty())
+	}
+
+	return metrics
+}
+
+func keySliceButDifferent(mapping map[string][]string) []string {
```

Comment:
The name is rather strange. Isn't it simple `getStringMapKeys`?

Reply:
I forgot to change it ^^ No overloading and no generics can sometimes be annoying. It's changed now.